src/context/QuestsContext.jsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import { useState } from 'react'
+
+const QuestContext = React.createContext(null)
+
+const DEFAULT_STATE = [
+	{
+		id: 1,
+		name: 'quest1',
+		isComplete: false,
+		isUnlocked: true,
+	},
+	{ id: 2, name: 'quest2', isComplete: false, isUnlocked: false },
+]
+
+const QuestProvider = ({ children }) => {
+	const getQuests = () => {
+		const questString = localStorage.getItem('quests')
+		const questMod = questString ? JSON.parse(questString) : DEFAULT_STATE
+		localStorage.setItem('quests', JSON.stringify(questMod))
+
+		return questMod
+	}
+
+	const getQuestCounter = () => {
+		const questString = localStorage.getItem('quest_number')
+		const questMod = questString ? JSON.parse(questString) : 1
+
+		localStorage.setItem('quest_number', JSON.stringify(questMod))
+
+		return questMod
+	}
+
+	const [quests, setQuests] = useState(getQuests())
+	const [questCounter, updateQuestCounter] = useState(getQuestCounter())
+
+	function markQuestAsCompleted() {

```

Comment:
Last time we discussed passing a quest id, which will mark as completed the quest.

Chanigng the idea to automatically detecting the quest that should be completed makes this more complicated, and no wonder why you had trouble finishing this up. It took me a while to figure out that "questCounter" is actually "currentlyActiveQuest" or something. Way too complicated in my opinion.